### PR TITLE
Fix update_config in ingress charm library

### PR DESCRIPTION
**Pull request by weiiwang01**
A bug was introduced in 1814172011d7fab9f0505f1bd6a11d6c07b8941a that causes updating `service-hostname`, `service-name`, `service-port` or `service-namespace` via the ingress charm library‘s `update_config` method has no effect.

Fixing this problem and add some integration tests for this situation.

### CODE REVIEW

File: .github/workflows/integration_test.yaml
Comment (mthaddon) [inline .github/workflows/integration_test.yaml:8]: Do we really want to switch to this, rather than landing whatever's in here in the main branch?

File: tests/integration/test_relation.py
Comment (mthaddon) [inline tests/integration/test_relation.py:64]: s/nad/and/

**Review by mthaddon — COMMENTED** [review]
Two comments inline

File: .github/workflows/integration_test.yaml
Comment (weiiwang01) [reply .github/workflows/integration_test.yaml:8]: No, this is just temporary to make the integration tests working, I will switch to the main branch after the [changes landed](https://github.com/canonical/operator-workflows/pull/50) in `operator-workflow` and before merge this PR.

**Review by gregory-schiano — COMMENTED** [review]
Had the same comment as Tom, I'll approve the PR once the changes on `operator-workflow`  are merged and the branch is back to `main` here :)

File: tests/integration/test_relation.py
Comment (weiiwang01) [reply tests/integration/test_relation.py]: Fixed, thanks!

File: tests/integration/test_relation.py
Comment (amandahla) [inline tests/integration/test_relation.py:44]: Is there no need for a "wait_for_idle" before the run action?

Comment (weiiwang01) [reply tests/integration/test_relation.py:45]: Fixed, thanks!

**Review by arturo-seijas — DISMISSED** [review]
LGTM

**Review by amandahla — DISMISSED** [review]
LGTM

File: .github/workflows/integration_test.yaml
Comment (jdkandersson) [inline .github/workflows/integration_test.yaml:11]: Final new line?

File: lib/charms/nginx_ingress_integrator/v0/ingress.py
Comment (jdkandersson) [inline lib/charms/nginx_ingress_integrator/v0/ingress.py:143]: Add argument and return value description?

File: lib/charms/nginx_ingress_integrator/v0/ingress.py
Comment (jdkandersson) [inline lib/charms/nginx_ingress_integrator/v0/ingress.py:269]: Final new line?

File: .github/workflows/integration_test.yaml
Comment (weiiwang01) [reply .github/workflows/integration_test.yaml]: Added, thanks!

File: tests/integration/test_relation.py
Comment (jdkandersson) [inline tests/integration/test_relation.py:35]: Should this be in a fixture?

File: tests/integration/test_relation.py
Comment (jdkandersson) [inline tests/integration/test_relation.py:50]: Should this depend on a fixture that builds and deploys the charm?

File: tests/integration/test_relation.py
Comment (jdkandersson) [inline tests/integration/test_relation.py:61]: Should this also depend on the fixture that builds and deploys the charm? Same comment for the remaining functions that depend on the deployed charm

File: lib/charms/nginx_ingress_integrator/v0/ingress.py
Comment (weiiwang01) [reply lib/charms/nginx_ingress_integrator/v0/ingress.py:143]: This is a charm library that will be used by many other charms. So I want to conform with the remaining code in this library and keep changes as minimal as possible.

File: lib/charms/nginx_ingress_integrator/v0/ingress.py
Comment (weiiwang01) [reply lib/charms/nginx_ingress_integrator/v0/ingress.py:269]: Added, thanks!

File: tests/integration/test_relation.py
Comment (weiiwang01) [reply tests/integration/test_relation.py:35]: Here is deploying the Nginx ingress charm and any-charm which is unique to this test module and not applicable to other test modules. Also, it's running everything concurrently to speed up the building and deploying process which makes it difficult to separate into fixtures. That's why I use a test case instead of a fixture.

File: tests/integration/test_relation.py
Comment (weiiwang01) [reply tests/integration/test_relation.py:50]: Same as above.

File: tests/integration/test_relation.py
Comment (weiiwang01) [reply tests/integration/test_relation.py:61]: Same as above.

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]
Looks great!

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]
That looks excellent now!

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**
